oneview/resource_server_profile_template.go

```diff
@@ -2306,9 +2315,11 @@ func resourceServerProfileTemplateRead(d *schema.ResourceData, meta interface{})
 					"name":                spt.LocalStorage.Controllers[i].LogicalDrives[j].Name,
 					"num_physical_drives": spt.LocalStorage.Controllers[i].LogicalDrives[j].NumPhysicalDrives,
 					"num_spare_drives":    spt.LocalStorage.Controllers[i].LogicalDrives[j].NumSpareDrives,
-					"sas_logical_jbod_id": spt.LocalStorage.Controllers[i].LogicalDrives[j].SasLogicalJBODId,
 					"raid_level":          spt.LocalStorage.Controllers[i].LogicalDrives[j].RaidLevel,
 				})
+				if spt.LocalStorage.Controllers[i].LogicalDrives[j].SasLogicalJBODId != 0 {
```

Comment:
in line 2309, we are assigning from OV and then here we are modifying it after converting into string. Can't we deal with it in line 2309 only

Reply:
Since LogicalDrives is a TypeSet, we must iterate and validate for each logicalDrives sas_logical_jbod_id attribute. So, we cant check it directly.

Reply:
"sas_logical_jbod_id": strconv.Itoa(spt.LocalStorage.Controllers[i].LogicalDrives[j].SasLogicalJBODId) will not work? only issue I can see adding of if condition in single line...generally ternary operators...but not sure in GoLang those are not simple....ok with current code

Reply:
@AvijitOS in case of an empty string, it will panic (because "" cant be converted to a number). Need an exception to bypass a panic.

Reply:
and that's true we can use ternary operator, but I am not sure if it will work while assigning value to a struct like below
``x := anySturct{ myIntStructVar: MyLocalVar > 0 ? 1 : 0 }``